File: cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    float change;
    float cchange;
    int rcents;

    do
    {
        printf("Change?");
        change = get_float();
    }

    while (change <= 0);

    //printf("input as a float is %.2f \n", change);

    //convert to int: multiply by 100 and round to an even cent

    //float to another float (original * 100)

    cchange = change * 100;

    //printf("times 100 is %.2f\n", cchange);

    rcents = round(cchange);

    //divide with quarters

    int quarter = 25;
    int count = 0;

    while (rcents >= quarter)
    {
        count ++;
        rcents = rcents - quarter;
    }

    int dime = 10;

    while (rcents >= dime)
    {
        count ++;
        rcents = rcents - dime;
    }

    int nickel = 5;

    while (rcents >= nickel)
    {
        count++;
        rcents = rcents - nickel;
    }

    int penny = 1;

    while (rcents >= penny)
    {
        count++;
        rcents = rcents - penny;

    }

    //output final total count

    printf("%i\n", count);

}

File: cash.py
# gives number of coins (25c 10c 5c 1c) to make a float input
import cs50

# get positive float input
while True:
    change = cs50.get_float("How much change?")
    if change > 0:
        break

# input in cents, rounded to nearest cent
cents = round(100 * change)

count = 0

# quarters, floor-divide by 25 for count, remainder is what's left!
count += (cents // 25)
cents = cents % 25

# dimes
count += (cents // 10)
cents = cents % 10

# nickels
count += (cents // 5)
cents = cents % 5

# pennies, no need to divide here.
count += cents

print(f"count is {count}")
